include/seqan3/argument_parser/validators.hpp

```diff
@@ -235,69 +236,87 @@ value_list_validator(std::vector<const char *> const & v) -> value_list_validato
 value_list_validator(std::initializer_list<const char *> const & v) -> value_list_validator<std::string>;
 //!\}
 
-/*!\brief A validator that checks if a filenames has one of the valid extensions.
+/*!\brief An abstract base class for the file and directory validators.
  * \ingroup argument_parser
- * \implements seqan3::validator_concept
  *
  * \details
  *
- * On construction, the validator must receive a list (vector) of valid file extensions.
- * The struct than acts as a functor, that throws a seqan3::parser_invalid_argument
- * exception whenever a given filename (string) is not in the given extension list.
- *
- * \snippet test/snippet/argument_parser/validators_3.cpp usage
- *
- * \note The validator works on every type that can be implicitly cast to std::filesystem::path.
+ * This class provides a common interface for seqan3::input_file_validator and the seqan3::output_file_validator as
+ * well as the seqan3::input_directory_validator and seqan3::output_directory_validator.
+ * This class is a CRTP-base class and cannot be instantiated directly. Furthermore, this is an exposition-only class.
+ * Please, do not use this class to derive from it as its interface might change without notice.
  */
-class file_ext_validator
+template <typename derived_t>
+class path_validator_base
```

Comment:
Why not normal polymorphy? (provide a protected `check_extension` member function that can be called when defining the `operator()` in the derived classes, then you do not need impl). I think CRTP is a little over-engineering for validators.. you even need to document explicitily to not use it. I'm not sure that it saves so much lines of code. I would even only provide a `check_extension` detail function and make no inheritance.

any way the base class should be detail?

Reply:
As discussed, I changed it now to an abstract base class with virtual function dispatching.